src/common/src/mlib/duration.h

```diff
@@ -0,0 +1,386 @@
+/**
+ * @file mlib/duration.h
+ * @brief Duration types and functions
+ * @date 2025-04-17
+ *
+ * This file contains types and functions for working with a "duration" type,
+ * which represents an elapsed amount of time, possibly negative.
+ *
+ * The type `mlib_duration_rep_t` is a typedef of the intregral type that is
+ * used to represent duration units.
+ *
+ * The `mlib_duration` is a trivial object that represents a duration of time.
+ * The internal representation should not be inspected outside of this file.
+ *
+ * @copyright Copyright (c) 2025
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
New files seem to have an auto-generated(?) doc-header format which does not match the license header format of other source files.

Reply:
It is semi-generated, but maintains the same copyright notice text below the `@copyright` block, although the `@copyright` line in these new files are wrong, so I'll fix that.